### PR TITLE
 Compile MSVC testers with warnings as errors

### PR DESCRIPTION
**Pull request by masterleinad**
Depends resolving the issue addressed in #10359.

### CODE REVIEW

File: .github/workflows/windows.yml [hunk 1]
```diff
@@ -8,6 +8,7 @@ jobs:
     name: Windows Serial
     runs-on: ${{ matrix.os }}
     strategy:
+      fail-fast: false
```
Comment (tjhei) [inline .github/workflows/windows.yml:11]: This will run both to completion, right? Is this useful to keep in?

Comment (masterleinad) [reply .github/workflows/windows.yml:11]: Yes, this avoids that other runs are aborted if one of them failed. I always found that confusing and normally I want to see all the (possible) errors at once.

**Comment by masterleinad** [comment]
Rebased.

File: .github/workflows/windows.yml (hunk 1 above)
Comment (tjhei) [reply .github/workflows/windows.yml:11]: makes sense.

**Pull request merged**